client/CPlayerInterface.cpp

```diff
@@ -267,8 +267,8 @@ void CPlayerInterface::heroMoved(const TryMoveHero & details, bool verbose)
 				assert(adventureInt->terrain.currentPath->nodes.size() >= 2);
 				std::vector<CGPathNode>::const_iterator nodesIt = adventureInt->terrain.currentPath->nodes.end() - 1;
 
-				if((nodesIt)->coord == CGHeroInstance::convertPosition(details.start, false)
-					&& (nodesIt - 1)->coord == CGHeroInstance::convertPosition(details.end, false))
+				if((nodesIt)->coord == hero->convertToVisitablePos(details.start)
+					&& (nodesIt - 1)->coord == hero->convertToVisitablePos(details.end))

```

Comment:
In some other changes you made, hero->convertToVisitablePos() was used when CGHeroInstance::convertPosition() 2nd argument was true rather than false, I hope this change doesn't add any missing offset

Reply:
Already double checked that, will do another round just in case. Correct change is:
```
convertPosition(pos, true) -> convertFromVisitablePos()
convertPosition(pos, false) -> convertToVisitablePos()
```

Reply:
False alert from my side